src/osx/cocoa/power.mm

```diff
@@ -105,3 +110,81 @@ bool UpdatePowerResourceUsage(wxPowerResourceKind kind, const wxString& reason)
             break;
     }
 }
+
+#import <Cocoa/Cocoa.h>
+#import "wx/power.h" // Assuming you have wxPowerEvent defined or a custom event
+
+@interface CocoaPowerEventsObserver : NSObject
```

Comment:
This probably should be called `wxCocoaPowerEventsObserver` for consistency and also maybe to avoid any conflicts with application-defined identifier when using static linking (I'm not sure what are the visibility rules for Objective C interfaces, but using "wx" prefix can never hurt).